syne_tune/optimizer/schedulers/multiobjective/random_scalarization.py

```diff
@@ -0,0 +1,94 @@
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, Optional, Iterable, List, Callable
+
+import numpy as np
+import scipy.stats
+from numpy.random import RandomState
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes import (
+    Predictor,
+    ScoringFunction,
+)
+from syne_tune.optimizer.schedulers.searchers.utils.common import Configuration
+
+
+class MultiObjectiveLCBRandomLinearScalarization(ScoringFunction):
+    """
+    Note: This is the multi objective random scalarization scoring function based on the work of Biswajit et al. [1].
+    This scoring function uses Lower Confidence Bound as the acquisition for the scalarized objective
+    :math:`h(\mu, \sigma) = \mu - \kappa * \sigma`
+
+        | [1] Paria, Biswajit, Kirthevasan Kandasamy and Barnabás Póczos.
+        | A Flexible Framework for Multi-Objective Bayesian Optimization using Random Scalarizations.
+        | Conference on Uncertainty in Artificial Intelligence (2018).
+
+    :param predictor: Surrogate predictor for statistics of predictive distribution
+    :param weights_sampler: Callable that can generate weights for each objective.
+        Once called it will return a dictionary mapping metric name to scalarization weight as
+        {
+            <name of metric 1> : <weight for metric 1>,
+            <name of metric 2> : <weight for metric 2>,
+            ...
+        }
+    """
+
+    def __init__(
+        self,
+        predictor: Dict[str, Predictor],
+        active_metric: Optional[List[str]] = None,
+        weights_sampler: Optional[Callable[[], Dict[str, float]]] = None,
+        kappa: float = 0.5,
+    ):
+        self.kappa = kappa
+        self.predictor = predictor
+        if active_metric is None:
+            active_metric = [name for name in predictor.keys()]
+        self.active_metric = active_metric
+
+        if weights_sampler is None:
+            state = RandomState(123)
+
+            def weights_sampler():
+                return {name: state.uniform() for name in predictor.keys()}
+
+        self.weights_sampler = weights_sampler
+
+    def score(
+        self,
+        candidates: Iterable[Configuration],
+        predictor: Optional[Dict[str, Predictor]] = None,
+    ) -> List[float]:
+
+        if predictor is None:
+            predictor = self.predictor
+        weights = self.weights_sampler()
+        scores = np.zeros(len(candidates))
+        for metric, metric_predictor in predictor.items():
+            predictions = metric_predictor.predict_candidates(candidates)
+            predicted_mean = predictions[0]["mean"]
+            predicted_std = predictions[0]["std"]
+
+            metric_score = self._single_objective_score(predicted_mean, predicted_std)
+            metrics_score_normalized = scipy.stats.rankdata(metric_score)
```

Comment:
Can you comment on why `rankdata` is used here? I do not get it.

Reply:
This is just rank normalization of the outputs, otherwise some objectives can be on different scale than others making it difficult to compare

Reply:
They use different normalization schemes. But their main point is that this framework is flexible so approaches like this should work very well. I have put it behind a flag and make it optional.